contracts/src/instances/ezeth/ezETHBase.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IERC20 } from "../../interfaces/IERC20.sol";
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { IRestakeManager } from "../../interfaces/IRestakeManager.sol";
+import { HyperdriveBase } from "../../internal/HyperdriveBase.sol";
+import { FixedPointMath, ONE } from "../../libraries/FixedPointMath.sol";
+
+/// @author DELV
+/// @title ezETH Base Contract
+/// @notice The base contract for the ezETH Hyperdrive implementation.
+/// @dev
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract EzETHBase is HyperdriveBase {
+    using FixedPointMath for uint256;
+
+    /// @dev The Renzo entrypoint contract.
+    IRestakeManager internal immutable _restakeManager;
+
+    /// @dev The ezETH token contract.
+    IERC20 internal immutable _ezETH;
+
+    /// @notice Instantiates the ezETH Hyperdrive base contract.
+    /// @param __restakeManager The Renzo Restakemanager contract.
+    constructor(IRestakeManager __restakeManager, IERC20 __ezETH) {
+        _restakeManager = __restakeManager;
+        _ezETH = __ezETH;
+        // TODO: see if we can get this to work so we don't need to pass in __ezETH
+        // (, bytes memory data) = address(__restakeManager).call(
+        //     abi.encodeWithSignature("ezETH()")
+        // );
+        // _ezETH = abi.decode(data, (IERC20));
```

Comment:
this should work if you decode as an address type

Reply:
yeah that worked, thanks!